src/components/ui/ImageGallery/ImageGallerySelector.tsx

```diff
@@ -34,14 +35,21 @@ const hasScroll = (container: HTMLDivElement | null): boolean => {
 
 function ImageGallerySelector({ images, onSelect, currentImageIdx }: Props) {
   const elementsRef = useRef<HTMLDivElement>(null)
+  const { ref: firstImageRef, inView: firstImageInView } = useInView({
+    threshold: 1,
+  })
+
+  const { ref: lastImageRef, inView: lastImageInView } = useInView({
+    threshold: 1,
+  })
```

Comment:
hmm, we can get the function return value previously as a const and use this const in the conditionals, preventing to call the function twice. 
Just remember to rename the function above to something like `selectorHasScroll` and replace the function call to the const value on the 2 calls.

```js
const hasScroll = selectorHasScroll(elementsRef.current)
```